---
layout: full
homepage: true
title: "DeMixT"
disable_anchors: false
description: Cell type-specific deconvolution of heterogeneous tumor samples with two or three components using expression data from RNAseq or microarray platforms
---

## DeMixT Overview

<!-- <div style="display:table; clear:both"> 
<div style="float: left;width: 50%;"> -->
Transcriptomic deconvolution in cancer and other heterogeneous tissues remains challenging. Available methods lack the ability to estimate both component-specific proportions and expression profiles for individual samples. We present DeMixT, a new tool to deconvolve high dimensional data from mixtures of more than two components. DeMixT implements an iterated conditional mode algorithm and a novel gene-set-based component merging approach to improve accuracy. In a series of experimental validation studies and application to TCGA data, DeMixT showed high accuracy. Improved deconvolution is an important step towards linking tumor transcriptomic data with clinical outcomes. An R package, scripts and data are available.

An example of DeMixT is [here](tutorial.html). 

<center>
<img src="./etc/demixt.jpg" alt="demixt" width="60%" />
</center>
<!-- </div> -->
<!-- <div style="float: right; width: 50%"> <img src="./etc/demixt.jpg" alt="demixt" /> </div> -->
<!-- </div> -->

